test/testLocaleData.js

```diff
@@ -0,0 +1,184 @@
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { setPlatform, getPlatform } from 'ilib-env';
+import { registerLoader } from 'ilib-loader';
+
+import MockLoader from './MockLoader';
+import LocaleData from '../src/LocaleData';
+
+module.exports.testLocaleData = {
+    testLocaleDataConstructor: function(test) {
+        test.expect(1);
+        const locData = new LocaleData({
+            path: "./test/files",
+            name: "test"
+        });
+        test.ok(locData);
+        test.done();
+    },
+    
+    testLocaleDataConstructorNoPath: function(test) {
+        test.expect(1);
+        test.throws((test) => {
+            new LocaleData({
+                name: "test"
+            });
+        });
+        test.done();
+    },
+    
+    testLocaleDataConstructorNoPackage: function(test) {
+        test.expect(1);
+        test.throws(() => {
+            new LocaleData({
+                path: "./test/files"
+            });
+        });
+        test.done();
+    },
+
+    testLocaleDataConstructorNoSync: function(test) {
+        test.expect(1);
+        const locData = new LocaleData({
+            path: "./test/files",
+            name: "test"
+        });
+        test.ok(!locData.isSync());
+
+        test.done();
+    },
+
+    testLocaleDataConstructorLoaderDoesntSupportSync: function(test) {
+        test.expect(1);
+        registerLoader(MockLoader);
+        setPlatform("mock");
+
+        const locData = new LocaleData({
+            path: "./test/files",
+            name: "test",
+            sync: true
+        });
+        test.ok(!locData.isSync());
+
+        // clean up
+        setPlatform(undefined);
+
+        test.done();
+    },
+
+    testLocaleDataNodeSyncRoot: function(test) {
+        setPlatform();
+
+        // only do this test on nodejs
+        if (getPlatform() !== "nodejs") {
+            test.done();
+            return;
+        }
+        test.expect(2);
+
+        const locData = new LocaleData({
+            path: "./test/files",
+            name: "test",
+            sync: true
+        });
+
+        test.ok(locData);
+        const actual = locData.loadData({
+            basename: "tester",
+            locale: "root"
+        });
+
+        test.equal(actual, {
+            "a": "b",
+            "c": "d",
+            "x": {
+               "m": "n",
+               "o": "p"
+            }
+        });
+        test.done();
+    },
+
+    testLocaleDataNodeSyncen: function(test) {
+        setPlatform();
+
+        // only do this test on nodejs
+        if (getPlatform() !== "nodejs") {
+            test.done();
+            return;
+        }
+        test.expect(2);
+
+        const locData = new LocaleData({
+            path: "./test/files",
+            name: "test",
+            sync: true
+        });
+
+        test.ok(locData);
+        const actual = locData.loadData({
```

Comment:
there are  6 assertion failures with the following message when I'm running `node test/testSuite.js` on local.
```
TypeError: Cannot read property 'data' of undefined
    at filter (/home/goun/Source/opensource_iLib/ilib-localedata/src/LocaleData.js:298:69)
    at Array.filter (<anonymous>)
    at forEach (/home/goun/Source/opensource_iLib/ilib-localedata/src/LocaleData.js:298:48)
    at Array.forEach (<anonymous>)
    at forEach (/home/goun/Source/opensource_iLib/ilib-localedata/src/LocaleData.js:297:19)
    at Array.forEach (<anonymous>)
    at LocaleData.loadData (/home/goun/Source/opensource_iLib/ilib-localedata/src/LocaleData.js:296:20)
    at Object.testLocaleDataNodeSyncen (/home/goun/Source/opensource_iLib/ilib-localedata/test/testLocaleData.js:136:32)
```